### ISSUE TITLE
MASM errors should track origin 

### ISSUE THREAD

**Issue by hackaugusto** [0xPolygonMiden/miden-vm#1298]
Context: https://github.com/0xPolygonMiden/miden-vm/issues/1295

When errors are generated, they should track their source location as in source file, line, and column, to make it easier to find the root cause.

**Comment by bobbinth** [comment]
I think compilation errors will have that as a part of #1277 (but @bitwalker can correct if I'm off here). Proving this info for runtime errors will take more work though.

**Comment by bitwalker** [comment]
Yep! During my refactoring, I made most source-related errors associated with a source span, so that those errors render a source code snippet when displayed. There are still errors that lack spans, but these can be addressed one-by-one after 1277 is merged.

**Comment by hackaugusto** [comment]
are we planning on making a new vm release? this is still a pain point, specially for error results that don't have error codes, like `NotBinaryValue`

**Comment by bobbinth** [comment]
> are we planning on making a new vm release? this is still a pain point, specially for error results that don't have error codes, like `NotBinaryValue`.

We are - but first we need to migrate to MAST-based program serialization. I think we should try to get this done (and the new version of the VM released) but early/mid June.

> Edit: Btw, question, it seems none of the `ExecutionError` variants have any context. Maybe these were not handled by #1277 ?

#1277 improved error reporting during the assembly phase only. Improving error reporting (i.e., adding source locations) during program execution is still outstanding and will probably be a separate task (which will happen after the next release).

**Comment by bitwalker** [comment]
Implemented in #1419. We do not yet have a canonical way to ship the sources a given MAST artifact was produced from, it is assumed for now that the sources are located at the same path (absolute or relative to the current working directory) as when it was compiled.

Once we have defined how to ship sources (and how the processor can access them), then I think we can update the VM to attach source spans to diagnostics (such as when assertions fail), and/or point to a file/line number. The Miden CLI can be updated even now to make use of source info when available, but I've held off on that until we have more clarity on how we expect to ship debug info.
